### PR TITLE
Update/deprecated code

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Every deprecated function has been removed and substituted with a newer one

### CODE REVIEW

File: ooniprobe/View/LogViewController.h [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (bassosimone) [inline ooniprobe/View/LogViewController.h]: <3

File: ooniprobe/View/SettingsTableViewController.mm [hunk 2]
```diff
@@ -226,6 +226,7 @@ -(void)reloadSettings {
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
     if (indexPath.section == 2 && indexPath.row == [advancedItems count] -1){
+        //NOT REMOVED, settings will be removed soon
```
Comment (bassosimone) [inline ooniprobe/View/SettingsTableViewController.mm]: 👍

**Comment by lorenzoPrimi** [comment]
🐳?

**Comment by bassosimone** [comment]
@lorenzoPrimi I am looking into this PR now.

File: ooniprobe/AppDelegate.mm [hunk 3]
```diff
@@ -25,7 +23,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     [Fabric with:@[[Crashlytics class]]];
     #endif
 
-    [[UIApplication sharedApplication] setStatusBarStyle:UIStatusBarStyleLightContent];
+    application.statusBarStyle = UIStatusBarStyleLightContent;
```
Comment (bassosimone) [inline ooniprobe/AppDelegate.mm]: Here you're using the property approach rather than the message dispatching one, correct?

File: ooniprobe/AppDelegate.mm [hunk 4]
```diff
@@ -77,20 +75,22 @@ -(void)handleNotification:(NSDictionary*)userInfo :(UIApplication *)application{
...
+                                       }];
             links = [[NSMutableArray alloc] init];
             [links addObject:[[userInfo objectForKey:@"payload"] objectForKey:@"href"]];
             if ([[userInfo objectForKey:@"payload"] objectForKey:@"alt_hrefs"]){
                 NSArray *alt_href = [[userInfo objectForKey:@"payload"] objectForKey:@"alt_hrefs"];
                 [links addObjectsFromArray:alt_href];
             }
-            alertView.tag = ALERT_TAG_NOTIFICATION;
-            [alertView show];
+            [MessageUtility alertWithTitle:nil message:[NSString stringWithFormat:@"%@",[[userInfo objectForKey:@"aps"] objectForKey:@"alert"]] okButton:okButton inView:self.window.rootViewController];
```
Comment (bassosimone) [inline ooniprobe/AppDelegate.mm]: Can you wrap this very long line here, please?

File: ooniprobe/AppDelegate.mm [hunk 5]
```diff
@@ -77,20 +75,22 @@ -(void)handleNotification:(NSDictionary*)userInfo :(UIApplication *)application{
...
         }
         else {
-            UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil message:[NSString stringWithFormat:@"%@",[[userInfo objectForKey:@"aps"] objectForKey:@"alert"]] delegate:nil cancelButtonTitle:NSLocalizedString(@"ok", nil) otherButtonTitles:nil];
-            [alertView show];
+            [MessageUtility alertWithTitle:nil message:[NSString stringWithFormat:@"%@",[[userInfo objectForKey:@"aps"] objectForKey:@"alert"]] inView:self.window.rootViewController];
```
Comment (bassosimone) [inline ooniprobe/AppDelegate.mm]: Can you wrap this very long line here, please?

File: ooniprobe/AppDelegate.mm [hunk 6]
```diff
@@ -191,9 +179,14 @@ -(void)handleUrlScheme:(NSURL*)url{
...
-            alertView.tag = ALERT_TAG_OPEN_ITUNES;
-            [alertView show];
+            UIAlertAction* okButton = [UIAlertAction
+                                       actionWithTitle:NSLocalizedString(@"ok", nil)
+                                       style:UIAlertActionStyleDefault
+                                       handler:^(UIAlertAction * action) {
+                                           NSString *iTunesLink = @"itms://itunes.apple.com/us/app/apple-store/id1199566366?mt=8";
+                                           [[UIApplication sharedApplication] openURL:[NSURL URLWithString:iTunesLink]];
+                                       }];
+            [MessageUtility alertWithTitle:NSLocalizedString(@"ooniprobe_outdate", nil) message:NSLocalizedString(@"ooniprobe_outdate_msg", nil) okButton:okButton inView:self.window.rootViewController];
```
Comment (bassosimone) [inline ooniprobe/AppDelegate.mm]: Can you wrap this very long line, please?

File: ooniprobe/AppDelegate.mm [hunk 7]
```diff
@@ -206,8 +199,7 @@ -(void)handleUrlScheme:(NSURL*)url{
                 if ([parameters objectForKey:@"tn"] && [[Tests currentTests] getTestWithName:[parameters objectForKey:@"tn"]])
                     [self openURIschemeScreen:parameters];
                 else {
-                    UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"invalid_parameter", nil) message:[NSString stringWithFormat:@"%@ : %@", NSLocalizedString(@"test_name", nil), [parameters objectForKey:@"tn"]] delegate:nil cancelButtonTitle:NSLocalizedString(@"Cancel", nil) otherButtonTitles:nil];
-                    [alertView show];
+                    [MessageUtility alertWithTitle:NSLocalizedString(@"invalid_parameter", nil) message:[NSString stringWithFormat:@"%@ : %@", NSLocalizedString(@"test_name", nil), [parameters objectForKey:@"tn"]] inView:self.window.rootViewController];
```
Comment (bassosimone) [inline ooniprobe/AppDelegate.mm]: Can you wrap this very long line, please?

File: ooniprobe/Utility/DictionaryUtility.m [hunk 8]
```diff
@@ -9,9 +9,8 @@ + (NSDictionary *)parseQueryString:(NSString *)query {
     
     for (NSString *pair in pairs) {
         NSArray *elements = [pair componentsSeparatedByString:@"="];
-        NSString *key = [[elements objectAtIndex:0] stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-        NSString *val = [[elements objectAtIndex:1] stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-        
+        NSString *key = [[elements objectAtIndex:0] stringByRemovingPercentEncoding];
+        NSString *val = [[elements objectAtIndex:1] stringByRemovingPercentEncoding];
```
Comment (bassosimone) [inline ooniprobe/Utility/DictionaryUtility.m]: What is the rationale of this change?

File: ooniprobe/Utility/MessageUtility.mm [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+                                 message:msg
+                                 preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction* cancelButton = [UIAlertAction
+                                   actionWithTitle:NSLocalizedString(@"cancel", nil)
+                                   style:UIAlertActionStyleDefault
+                                   handler:nil];
+    [alert addAction:cancelButton];
+    [alert addAction:okButton];
+    [view presentViewController:alert animated:YES completion:nil];
+}
```
Comment (bassosimone) [inline ooniprobe/Utility/MessageUtility.mm]: Isn't it possible here to refactor code such that one of the two functions calls the other?

File: ooniprobe/View/PastTestsViewController.mm [hunk 10]
```diff
@@ -180,8 +180,7 @@ -(void)goToTest:(NetworkMeasurement*)current{
     else if ([items count] == 1 && [[items objectAtIndex:0] length] > 0)
         [self performSegueWithIdentifier:@"toResult" sender:self];
     else {
-        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"no_result", nil) message:NSLocalizedString(@"no_result_msg", @"") delegate:self cancelButtonTitle:NSLocalizedString(@"ok", @"") otherButtonTitles:nil];
-        [alert show];
+        [MessageUtility alertWithTitle:NSLocalizedString(@"no_result", nil) message:NSLocalizedString(@"no_result_msg", nil) inView:self];
```
Comment (bassosimone) [inline ooniprobe/View/PastTestsViewController.mm]: Can you wrap this very long line please?

File: ooniprobe/View/PastTestsViewController.mm [hunk 11]
```diff
@@ -209,16 +208,14 @@ - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEd
...
-        [self reloadTable];
-    }
+    UIAlertAction* okButton = [UIAlertAction
+                               actionWithTitle:NSLocalizedString(@"ok", nil)
+                               style:UIAlertActionStyleDefault
+                               handler:^(UIAlertAction * action) {
+                                   [TestStorage remove_all_tests];
+                                   [self reloadTable];
+                               }];
+    [MessageUtility alertWithTitle:NSLocalizedString(@"clear_all_tests_alert", nil) message:nil okButton:okButton inView:self];
```
Comment (bassosimone) [inline ooniprobe/View/PastTestsViewController.mm]: Can you wrap this line, please?

File: ooniprobe/View/ResultViewController.mm [hunk 12]
```diff
@@ -80,16 +80,13 @@ - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigati
...
-{
-    if (buttonIndex == 1) {
-        [[UIApplication sharedApplication] openURL:openURL];
+        UIAlertAction* okButton = [UIAlertAction
+                                   actionWithTitle:NSLocalizedString(@"ok", nil)
+                                   style:UIAlertActionStyleDefault
+                                   handler:^(UIAlertAction * action) {
+                                       [[UIApplication sharedApplication] openURL:openURL];
+                                   }];
+        [MessageUtility alertWithTitle:NSLocalizedString(@"open_url_alert", nil) message:url okButton:okButton inView:self];
```
Comment (bassosimone) [inline ooniprobe/View/ResultViewController.mm]: Can you wrap this line, please?

File: ooniprobe/Utility/DictionaryUtility.m (hunk 8 above)
Comment (lorenzoPrimi) [reply ooniprobe/Utility/DictionaryUtility.m]: stringByReplacingPercentEscapesUsingEncoding was deprecated

File: ooniprobe/Utility/MessageUtility.mm (hunk 9 above)
Comment (lorenzoPrimi) [reply ooniprobe/Utility/MessageUtility.mm]: No because, the okButton has no action in one function, while when there are two buttons the cancel button is the one with no action.
I want to keep as less parameters as possible.

Comment (bassosimone) [reply ooniprobe/Utility/MessageUtility.mm]: ok
